### ISSUE TITLE
Get Nan as training loss when using a Panoptic model

### ISSUE THREAD

**Issue by jgu13** [vanvalenlab/deepcell-tf#657]
**Problem description:**
I am using `Training a segmentation model.ipynb` as a template to segment my data. I have been testing using a small sample of the tissuenet_v1.1_test.npz dataset I downloaded from [https://datasets.deepcell.org/data](https://datasets.deepcell.org/data) This sample data can be found under data/minitest/X.npy and data/minitest/y.npy. These data is produced using the script `Process_tissuenetv1.py` where only nuclei images were retained. 

**Steps to reproduce the behavior**
Run the following script:
```
from parse_args import parse_args
import json
from tensorflow.keras.optimizers import SGD, Adam
from deepcell.model_zoo.panopticnet import PanopticNet
from deepcell.utils.train_utils import rate_scheduler
from deepcell.utils.train_utils import get_callbacks
from deepcell.utils.train_utils import count_gpus
from deepcell import image_generators
from deepcell.utils import train_utils
from deepcell import losses
from tensorflow.keras.losses import MSE
import os
import numpy as np
import tensorflow as tf
import time


def semantic_loss(n_classes):
    def _semantic_loss(y_true, y_pred):
        if n_classes > 1:
            return 0.01 * losses.weighted_categorical_crossentropy(
                y_true, y_pred, n_classes=n_classes)
        return MSE(y_true, y_pred)
    return _semantic_loss

def add_loss_to_semantic_heads(model):
    loss = {}
    # Give losses for all of the semantic heads
    for layer in model.layers:
        if layer.name.startswith('semantic_'):
            n_classes = layer.output_shape[-1]
            loss[layer.name] = semantic_loss(n_classes) # each layer has an independent loss
    return loss

def load_data(train_set_path, val_set_path, test_set_path):
    train_data = np.load(train_set_path)
    val_data = np.load(val_set_path)
    test_data = np.load(test_set_path)
    X_train, y_train = train_data['X'], train_data['y']
    X_val, y_val = val_data['X'], val_data['y']
    X_test, y_test = test_data['X'], test_data['y']
    return X_train, X_val, X_test, y_train, y_val, y_test

def main():
    backbone = 'resnet50'
    train_set_path = "/data/myData/mini_train.npz"
    val_set_path = "/data/myData/mini_val.npz"
    test_set_path = "/data/myData/mini_test.npz"
    model_dir  = "models"
    model_name = "mini_test"
    log_dir = "logs"

    X_train, X_val, _, y_train, y_val, _ = load_data(train_set_path, val_set_path, test_set_path)

    classes = {
        # inner_distance : 1,
        # outer_distance: 1,
        "fgbg" : 2
        }
    model = PanopticNet(
        backbone=backbone,
        input_shape=X_train.shape[1:],
        norm_method=None,
        num_semantic_classes=classes,
        location=True,
        include_top=False)
    
    optimizer = Adam(learning_rate=1e-4, clipnorm=0.001)
    lr_sched = rate_scheduler(lr=1e-4, decay=0.99)
    batch_size=10
    min_objects=1
    transforms = list(classes.keys())
    transforms_kwargs = {'inner-distance': {'erosion_width': 0}}
    seed = 42

    # use augmentation for training but not validation
    datagen = image_generators.SemanticDataGenerator(
        rotation_range=180,
        shear_range=0,
        zoom_range=(0.75, 1.25),
        horizontal_flip=True,
        vertical_flip=True)

    datagen_val = image_generators.SemanticDataGenerator(
        rotation_range=0,
        shear_range=0,
        zoom_range=0,
        horizontal_flip=0,
        vertical_flip=0)
        
    train_data = datagen.flow(
        {'X': X_train, 'y': y_train},
        seed=seed,
        transforms=transforms,
        transforms_kwargs=transforms_kwargs,
        min_objects=min_objects,
        batch_size=batch_size)

    val_data = datagen_val.flow(
        {'X': X_val, 'y': y_val},
        seed=seed,
        transforms=transforms,
        transforms_kwargs=transforms_kwargs,
        min_objects=min_objects,
        batch_size=batch_size)

    loss = add_loss_to_semantic_heads(model)
    model.compile(loss=loss, optimizer=optimizer)


    model_path = os.path.join(model_dir, '{}.h5'.format(model_name))

    num_gpus = count_gpus()

    print('Training on', num_gpus, 'GPUs.')

    train_callbacks = get_callbacks(
    model_path,
    lr_sched=lr_sched,
    tensorboard_log_dir=log_dir,
    save_weights_only=num_gpus >= 2,
    monitor='val_loss',
    verbose=1)

    loss_history = model.fit(
    train_data,
    steps_per_epoch=train_data.y.shape[0] // batch_size,
    epochs=10,
    validation_data=val_data,
    validation_steps=val_data.y.shape[0] // batch_size,
    callbacks=train_callbacks)


if __name__ == '__main__':
    main()
```

**Expected behavior**
Not getting nan as loss.

**Current behavior**
When training starts, only the loss values of semantic_inner and semantic_outer layers from the first epoch are not nan values. 
Current output
```
Training on 1 GPUs.

 1/24 [>.............................] - ETA: 7:33 - loss: nan
 2/24 [=>............................] - ETA: 2s - loss: nan  
 3/24 [==>...........................] - ETA: 3s - loss: nan
 4/24 [====>.........................] - ETA: 3s - loss: nan
 5/24 [=====>........................] - ETA: 3s - loss: nan
 6/24 [======>.......................] - ETA: 3s - loss: nan
 7/24 [=======>......................] - ETA: 3s - loss: nan
 8/24 [=========>....................] - ETA: 3s - loss: nan
 9/24 [==========>...................] - ETA: 3s - loss: nan
10/24 [===========>..................] - ETA: 2s - loss: nan
11/24 [============>.................] - ETA: 2s - loss: nan
12/24 [==============>...............] - ETA: 2s - loss: nan
13/24 [===============>..............] - ETA: 2s - loss: nan
14/24 [================>.............] - ETA: 2s - loss: nan
15/24 [=================>............] - ETA: 1s - loss: nan
16/24 [===================>..........] - ETA: 1s - loss: nan
17/24 [====================>.........] - ETA: 1s - loss: nan
18/24 [=====================>........] - ETA: 1s - loss: nan
19/24 [======================>.......] - ETA: 1s - loss: nan
20/24 [========================>.....] - ETA: 0s - loss: nan
21/24 [=========================>....] - ETA: 0s - loss: nan
22/24 [==========================>...] - ETA: 0s - loss: nan
23/24 [===========================>..] - ETA: 0s - loss: nan
24/24 [==============================] - ETA: 0s - loss: nan
Epoch 1: val_loss did not improve from inf
24/24 [==============================] - 26s 289ms/step - loss: nan - val_loss: nan - lr: 1.0000e-04
```

**Desktop (please complete the following information):**
 - Python Version [e.g. 3.9.6]

**Comment by rossbar** [comment]
From the run log, the problem looks like it lies in the `semantic_fgbg_loss`. It'd be helpful if you could provide a minimal reproducing example (i.e. try to distill the components from your `scripts/main.py` that are necessary to reproduce the error); otherwise I'd start by focusing on the fgbg loss function as applied to this data.

**Comment by jgu13** [comment]
Hi @rossbar , thank you for your response. I have updated the question with a minimal reproducible example. The nan loss only occurs when the semantic heads are set to `{'fgbg' : 2}`.

**Comment by jgu13** [comment]
Hi, I have used `tf.losses.CategoricalCrossEntropy` as the loss function and the nan problem goes away. But now my trained model does not predict class of each pixel correctly. The last softmax layer only outputs very high probabilities to the foreground class and very low probabilities to the background class.

**Comment by jgu13** [comment]
Also, how can I load weights that I got from pre-training a panoptic model on a sample tissuenet dataset before training the same model on my own dataset? Thanks!
